### PR TITLE
Validation for admin.email config when sending ToolchainStatus notification

### PR DESCRIPTION
**Pull request by sbryzak**
Fixes https://issues.redhat.com/browse/CRT-881

Signed-off-by: Shane Bryzak <sbryzak@gmail.com>

### CODE REVIEW

File: pkg/controller/toolchainstatus/toolchainstatus_controller.go [hunk 1]
```diff
@@ -351,6 +354,10 @@ func (r *ReconcileToolchainStatus) membersHandleStatus(reqLogger logr.Logger, to
 func (r *ReconcileToolchainStatus) sendToolchainStatusUnreadyNotification(logger logr.Logger,
 	toolchainStatus *toolchainv1alpha1.ToolchainStatus) error {
 
+	if !isValidEmailAddress(r.config.GetAdminEmail()) {
+		return errs.New("cannot create notification due to configuration error - admin.email is invalid or not set")
```
Comment (alexeykazakov) [inline pkg/controller/toolchainstatus/toolchainstatus_controller.go:358]: Let's add the problematic email to the error message

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go [hunk 2]
```diff
@@ -763,6 +763,44 @@ func TestToolchainStatusNotifications(t *testing.T) {
...
 			assertToolchainStatusNotificationNotCreated(t, fakeClient)
 
+			t.Run("Notification not created when admin.email not configured", func(t *testing.T) {
+				invalidConfig := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notification_test_config",
+						Namespace: test.HostOperatorNs,
+					},
+					Data: map[string]string{
+						"admin.email": "",
```
Comment (alexeykazakov) [inline pkg/controller/toolchainstatus/toolchainstatus_controller_test.go:773]: Need to add a test for not empty but invalid email.

File: pkg/controller/toolchainstatus/toolchainstatus_controller.go (hunk 1 above)
Comment (sbryzak) [reply pkg/controller/toolchainstatus/toolchainstatus_controller.go]: Done

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go (hunk 2 above)
Comment (sbryzak) [reply pkg/controller/toolchainstatus/toolchainstatus_controller_test.go:773]: Done

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go [hunk 3]
```diff
@@ -763,6 +763,84 @@ func TestToolchainStatusNotifications(t *testing.T) {
...
+				// then
+				require.Error(t, err)
+				require.Equal(t, "Failed to create user deactivation notification: cannot create notification "+
+					"due to configuration error - admin.email [admin#foo.com] is invalid or not set",
+					err.Error())
+				assert.Equal(t, requeueResult, res)
+
+				// Confirm there is no notification
+				assertToolchainStatusNotificationNotCreated(t, fakeClient)
+			})
```
Comment (alexeykazakov) [inline pkg/controller/toolchainstatus/toolchainstatus_controller_test.go:842]: If I'm not missing something these two tests are almost fully identical. The only difference is the email.
You could create a custom assertion function to re-use it in the tests. Similar https://github.com/codeready-toolchain/member-operator/blob/798f0752ce6c46490cf9db2057c0d0b814dc62ac/pkg/controller/idler/idler_controller_test.go#L263-L283

**Review by alexeykazakov — APPROVED** [review]
Looks good! But the tests could be slightly refactored. See the comment :)

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go (hunk 3 above)
Comment (sbryzak) [reply pkg/controller/toolchainstatus/toolchainstatus_controller_test.go:810]: Refactored.
